mlir/lib/Dialect/MIOpen/Pipelines/XMIRPipelines.cpp

```diff
@@ -53,15 +40,23 @@ void xmir::buildModelPipeline(OpPassManager &pm,
 
 void xmir::buildRunnerPipeline(OpPassManager &pm,
                                const xmir::RunnerOptions &options) {
+  pm.addNestedPass<func::FuncOp>(createConvertLinalgToAffineLoopsPass());
   pm.addPass(createLowerAffinePass());
   pm.addPass(createConvertSCFToCFPass());
-  if (!options.cpuOnly) {
+  if (options.cpuOnly) {
+
+  } else {
     pm.addPass(createConvertAsyncToGPUPass());
     pm.addPass(createSymbolDCEPass());
+    pm.addPass(createGpuToLLVMConversionPass());
+    pm.addNestedPass<func::FuncOp>(createGpuAsyncRegionPass());
   }
   pm.addNestedPass<func::FuncOp>(createConvertMathToLLVMPass());
-  pm.addPass(createGpuToLLVMConversionPass());
+  pm.addPass(createConvertLinalgToLLVMPass());
```

Comment:
Shouldn't these guys be taking an LLVM type converter so that we can set index-bitwidth=32?

Reply:
This isn't for the kernel generation, just host code.